chainer/functions/pooling/roi_max_align_2d.py

```diff
@@ -0,0 +1,497 @@
+                (roi_batch_ind * channels + c) * height * width;
+
+            int top_offset = (n * channels + c) * pooled_height * pooled_width;
+            T top_diff_this_bin =
+                top_diff[top_offset + ph * pooled_width + pw];
+
+            // We use roi_bin_grid to sample the grid and mimic integral
+            int roi_bin_grid_h = (sampling_ratio_h > 0)
+                ? sampling_ratio_h
+                : ceil(roi_height / pooled_height); // e.g. = 2
+            int roi_bin_grid_w = (sampling_ratio_w > 0)
+                ? sampling_ratio_w
+                : ceil(roi_width / pooled_width);
+
+            int max_index = argmax_data[top_offset + ph * pooled_width + pw];
+            int iy = max_index / roi_bin_grid_w;
+            int ix = max_index % roi_bin_grid_w;
+
+            T y = roi_start_h + ph * bin_size_h +
+                static_cast<T>(iy + .5f) * bin_size_h /
+                    static_cast<T>(roi_bin_grid_h);  // e.g. 0.5, 1.5
+            T x = roi_start_w + pw * bin_size_w +
+                static_cast<T>(ix + .5f) * bin_size_w /
+                    static_cast<T>(roi_bin_grid_w);
+
+            // bilinear_interpolation_gradient {{
+            int y_low, x_low, y_high, x_high;
+            T w1, w2, w3, w4;
+            bool ret = get_bilinear_interp_params(
+                x, y, height, width,
+                y_low, x_low, y_high, x_high,
+                w1, w2, w3, w4
+            );
+            if (!ret) {
+                continue;
+            }
+
+            T g1 = top_diff_this_bin * w1;
+            T g2 = top_diff_this_bin * w2;
+            T g3 = top_diff_this_bin * w3;
+            T g4 = top_diff_this_bin * w4;
+
+            if (x_low >= 0 && x_high >= 0 &&
+                    y_low >= 0 && y_high >= 0) {
+                atomicAdd(&bottom_diff[bottom_diff_offset +
+                                       y_low * width + x_low], g1);
+                atomicAdd(&bottom_diff[bottom_diff_offset +
+                                       y_low * width + x_high], g2);
+                atomicAdd(&bottom_diff[bottom_diff_offset +
+                                       y_high * width + x_low], g3);
+                atomicAdd(&bottom_diff[bottom_diff_offset +
+                                       y_high * width + x_high], g4);
+            }
+            // }}
+            ''',
+            'roi_max_align_2d_bwd',
+            preamble=_GET_BILINEAR_INTERP_KERNEL,
+        )(gy[0], self.spatial_scale, channels, height, width,
+          self.outh, self.outw, sampling_ratio_h, sampling_ratio_w,
+          bottom_rois, bottom_roi_indices, bottom_diff, self.argmax_data,
+          size=gy[0].size)
+
+        return bottom_diff, None, None
+
+
+def roi_max_align_2d(
+        x, rois, roi_indices, outsize, spatial_scale, sampling_ratio=None
+):
+    """Spatial Region of Interest (ROI) max align function.
+
+    This function acts similarly to :class:`~functions.ROIPooling2D`, but
```

Comment:
`functions.ROIPooling2D` was removed and `functions.roi_pooling_2d` is deprecated. How about pointing `functions.roi_max_pooling_2d` or `functions.roi_average_align_2d`?

Reply:
I want to refer `functions.roi_max_poolind_2d`, but it is not merged yet :(

Reply:
#5304 is merged now :tada: :tada: :tada: 
Could you merge master into this PR and update the docs?

Reply:
Updated :innocent: